anchore_engine/services/policy_engine/api/controllers/feeds.py

```diff
@@ -153,23 +130,23 @@ def sync_feeds(sync=True, force_flush=False):
 def toggle_feed_enabled(feed, enabled):
     if type(enabled) != bool:
         raise BadRequest(message="state must be a boolean", detail={"value": enabled})
-    session = db.get_session()
+
     try:
-        f = db.set_feed_enabled(session, feed, enabled)
-        if not f:
+        provider = get_vulnerabilities_provider()
+        feed = provider.toggle_feed_enabled(feed, enabled)
+
+        if not feed:
             raise ResourceNotFound(feed, detail={})
-        session.flush()
 
-        updated = _marshall_feed_response(f).to_json()
-        session.commit()
+        return jsonify(feed.to_json()), 200
 
-        return jsonify(updated), 200
-    except AnchoreApiError:
-        session.rollback()
-        raise
+    except InvalidFeed:
+        raise BadRequest(
+            message="Feed not enabled on configured vulnerability provider",
+            detail={"feed": feed, "configured_provider": provider.__config__name__},
```

Comment:
use `provider.get_config_name()` The error message is a bit confusing. It should be something to the effet of "Feed not supported"